### PR TITLE
Merge rubocop configuration: 2nd PR

### PR DESCRIPTION
**Pull request by hsbt**
## What was the end-user or developer problem that led to this PR?

This is second migration work of https://github.com/rubygems/rubygems/pull/6487. 

## What is your fix for the problem, implemented in this PR?

I migrate bundler's cops to rubygems with following conditions:

* Without effect of Bundler
* Without test failure
* Without change of logic meanings

Finally, I choose and apply safety Cops with auto-correct. 

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Comment by indirect** [comment]
Thank you for doing this work! 🙌🏻

**Comment by deivid-rodriguez** [comment]
@hsbt Thanks! I love it that you splitted commits. The way I do backports for releases, when conflicts happen, I normally also backport the change that originated the conflict in order to resolve it if it's harmless. Being able to backport just commits with focused style changes should make things easier.

If you don't mind, I'd like to tweak it a bit more and do a rebase pass to remove "rollback commits", and to include moving each cop from `.rubocop_bundler.yml` to `.rubocop.yml` with each commit that fixes the offenses for that cop.

**Comment by hsbt** [comment]
> If you don't mind, I'd like to tweak it a bit more and do a rebase pass to remove "rollback commits", and to include moving each cop from `.rubocop_bundler.yml` to `.rubocop.yml` with each commit that fixes the offenses for that cop.

I understood your concern about backport commits. I'll re-organize this PR each cops rule and its change.

**Comment by hsbt** [comment]
@deivid-rodriguez done. I added cop change to each commit. But It may causes conflicts when you do backport 🙏

**Comment by hsbt** [comment]
I'm not sure why GitHub still use old status check contained old versions of macOS platforms. I will merge this without merge queue.

**Comment by hsbt** [comment]
@deivid-rodriguez @simi Sorry, I disabled and enabled status check of branch protection rule. After that, the current configuration was empty. I added the following rules again.

* Bundler 2 tests on Ubuntu, Windows, macOS
* Realworld Bundler 2 tests on Ubuntu, macOS
* Bundler 3 tests on Ubuntu
* Realworld Bundler 3 on Ubuntu
* RubyGems tests on Ubuntu, Windows, macOS
* Bundler and Rubygems tests for ruby-core

If you remember other rules, can you add it to branch protection rule by yourself?

**Comment by deivid-rodriguez** [comment]
Mmmm, I think we have to feature to allow maintainers merge regardless of CI status? You click a red button, but if you're a maintainer, you can click it. So next time, I think you don't need to disable & enable again status checks (specially if we'll lose all required checks 😆). Anyways, it's no big deal, I can review if there's anything missing on another PR.

What's important here is... This PR is awesome, thanks so much! No more CI split-brain and lint CI complaints because using different styles in the two projects! 💪

File: test/rubygems/test_gem_specification.rb [hunk 1]
```diff
@@ -804,8 +804,7 @@ def test_self_load_tainted
       end
 
       full_path.taint
-      loader = Thread.new { $SAFE = 1
- Gem::Specification.load full_path }
+      loader = Thread.new { $SAFE = 1; Gem::Specification.load full_path }
```
Comment (deivid-rodriguez) [inline test/rubygems/test_gem_specification.rb]: A glitch in the system 😆. Hopefully not a bug in `util/rubocop -A --only Style/BarePercentLiterals` 😅. Luckily it got corrected back in a later commit :)

**Pull request merged**